widgetry/src/text.rs

```diff
@@ -441,32 +446,24 @@ impl Text {
 }
 
 fn render_line(spans: Vec<TextSpan>, tolerance: f32, assets: &Assets) -> GeomBatch {
-    // TODO This assumes size and font don't change mid-line. We might be able to support that now,
-    // actually.
-    // https://www.oreilly.com/library/view/svg-text-layout/9781491933817/ch04.html
-
     // Just set a sufficiently large view box
     let mut svg = r##"<svg width="9999" height="9999" viewBox="0 0 9999 9999" xmlns="http://www.w3.org/2000/svg">"##.to_string();
 
-    write!(
-        &mut svg,
-        r##"<text x="0" y="0" xml:space="preserve" font-size="{}" font-family="{}" {}>"##,
-        spans[0].size,
-        spans[0].font.family(),
-        match spans[0].font {
-            Font::OverpassBold => "font-weight=\"bold\"",
-            Font::OverpassSemiBold => "font-weight=\"600\"",
-            _ => "",
-        }
-    )
-    .unwrap();
+    write!(&mut svg, r##"<text x="0" y="0" xml:space="preserve">"##,).unwrap();
 
     let mut contents = String::new();
     for span in spans {
         let fg_color = span.fg_color_for_style(&assets.style.borrow());
         write!(
             &mut contents,
-            r##"<tspan fill="{}" fill-opacity="{}" {}>{}</tspan>"##,
+            r##"<tspan font-size="{}" font-family="{}" {} fill="{}" fill-opacity="{}" {}>{}</tspan>"##,
```

Comment:
This was... comically easy to support. I struggled a fair bit to get text rendering at all originally; it's pretty refreshing to come back to this code and fix something so easily.